locallibrary/catalog/models.py

```diff
@@ -1,3 +1,92 @@
 from django.db import models
 
 # Create your models here.
+
+class Genre(models.Model):
+	"""Model representing a book genre."""
+	name = models.CharField(max_length=200, help_text="Enter a book genre (e.g.Science Fiction)")
+
+	def __str__(self):
+		"""String for representing the Model object."""
+		return self.name
+
+# Book model
+from django.urls import reverse 
+# Used to generate URLs by reversing the URL patternsclass 
+
+class Book(models.Model):
+	"""Model representing a book (but not a specific copy of a book)."""
+	title = models.CharField(max_length=200)
+
+	author = models.ForeignKey("Author", on_delete=models.SET_NULL, null=True)
```

Comment:
Tìm hiểu thêm các giá trị khác của option `on_delete` e nha

Reply:
Hiện tại thì tóm tắt của em như thế này ạ:

#define "get reckt" "get delete"
**CASCADE**: emulate ON DELETE CASCADE, also make obj containing Fo.Key get reckt
**PROTECT**: prevent get reckt, raise ProtectedError
**RESTRICT**: raise RestrictedError, only allow deletion if referenced also get reckt via CASCADE
**SET_NULL**: if null == True then Fo.Key = null
**SET_DEFAULT**: need to set FroreignKey default value before use
**SET()**: set ForeignKey as the value that passed in SET() (variable, funct,…), usually callable stuff to not invoke querries when import models.py
**DO_NOTHING**: nop, pass. Cause **IntegrityError** if backend enforces, have to mannualy add an SQL **ON DELETE** constraint to the db field